src/Util/ClassSourceManipulator.php

```diff
@@ -121,13 +121,23 @@ public function addEntityField(ClassProperty $mapping): void
 
         $defaultValue = null;
         $commentLines = [];
-        if ('array' === $typeHint && !$nullable) {
-            $defaultValue = new Node\Expr\Array_([], ['kind' => Node\Expr\Array_::KIND_SHORT]);
-            if (null !== $mapping->enumType) {
+
+        if (null !== $mapping->enumType) {
+            if ('array' === $typeHint) {
+                // still need to add the use statement
+                $this->addUseStatementIfNecessary($mapping->enumType);
+
                 $commentLines = [\sprintf('@return %s[]', Str::getShortClassName($mapping->enumType))];
+                if ($nullable) {
+                    $commentLines[0] .= '|null';
```

Comment:
To be consistent in the bundle, lets use `sprintf()`

```suggestion
                    $commentLines[0] = \sprintf('%s|null', $commentLines[0]);
```